core/embed/extmod/modtrezorio/modtrezorio-fatfs.h

```diff
@@ -0,0 +1,524 @@
+    [FR_DISK_ERR] = MP_EIO,
+    [FR_INT_ERR] = MP_EIO,
+    [FR_NOT_READY] = MP_EBUSY,
+    [FR_NO_FILE] = MP_ENOENT,
+    [FR_NO_PATH] = MP_ENOENT,
+    [FR_INVALID_NAME] = MP_EINVAL,
+    [FR_DENIED] = MP_EACCES,
+    [FR_EXIST] = MP_EEXIST,
+    [FR_INVALID_OBJECT] = MP_EINVAL,
+    [FR_WRITE_PROTECTED] = MP_EROFS,
+    [FR_INVALID_DRIVE] = MP_ENODEV,
+    [FR_NOT_ENABLED] = MP_ENODEV,
+    [FR_NO_FILESYSTEM] = MP_ENODEV,
+    [FR_MKFS_ABORTED] = MP_EIO,
+    [FR_TIMEOUT] = MP_EIO,
+    [FR_LOCKED] = MP_EIO,
+    [FR_NOT_ENOUGH_CORE] = MP_ENOMEM,
+    [FR_TOO_MANY_OPEN_FILES] = MP_EMFILE,
+    [FR_INVALID_PARAMETER] = MP_EINVAL,
+};
+
+STATIC mp_obj_t filinfo_to_tuple(const FILINFO *info) {
+  mp_obj_tuple_t *tuple = MP_OBJ_TO_PTR(mp_obj_new_tuple(3, NULL));
+  tuple->items[0] = mp_obj_new_int_from_uint(info->fsize);
+  char attrs[] = "-----";
+  if (info->fattrib & AM_RDO) {
+    attrs[0] = 'r';
+  }
+  if (info->fattrib & AM_HID) {
+    attrs[1] = 'h';
+  }
+  if (info->fattrib & AM_SYS) {
+    attrs[2] = 's';
+  }
+  if (info->fattrib & AM_DIR) {
+    attrs[3] = 'd';
+  }
+  if (info->fattrib & AM_ARC) {
+    attrs[4] = 'a';
+  }
+  tuple->items[1] = mp_obj_new_str(attrs, 5);
+  tuple->items[2] = mp_obj_new_str(info->fname, strlen(info->fname));
+  return MP_OBJ_FROM_PTR(tuple);
+}
+
+/// class FatFSFile:
+///     """
+///     Class encapsulating file
+///     """
+typedef struct _mp_obj_FatFSFile_t {
+  mp_obj_base_t base;
+  FIL fp;
+} mp_obj_FatFSFile_t;
+
+/// def __exit__(self) -> None:
+///     """
+///     Close an open file object
+///     """
+STATIC mp_obj_t mod_trezorio_FatFSFile___exit__(size_t n_args,
+                                                const mp_obj_t *args) {
+  mp_obj_FatFSFile_t *o = MP_OBJ_TO_PTR(args[0]);
+  FRESULT res = f_close(&(o->fp));
+  if (res) {
+    mp_raise_OSError(fresult_to_errno_table[res]);
+  }
+  return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(mod_trezorio_FatFSFile___exit___obj,
+                                           4, 4,
+                                           mod_trezorio_FatFSFile___exit__);
+
+/// def close(self) -> None:
+///     """
+///     Close an open file object
+///     """
+STATIC mp_obj_t mod_trezorio_FatFSFile_close(mp_obj_t self) {
+  mp_obj_FatFSFile_t *o = MP_OBJ_TO_PTR(self);
+  FRESULT res = f_close(&(o->fp));
+  if (res) {
+    mp_raise_OSError(fresult_to_errno_table[res]);
+  }
+  return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(mod_trezorio_FatFSFile_close_obj,
+                                 mod_trezorio_FatFSFile_close);
+
+/// def read(self, data: bytearray) -> int:
+///     """
+///     Read data from the file
+///     """
+STATIC mp_obj_t mod_trezorio_FatFSFile_read(mp_obj_t self, mp_obj_t data) {
+  mp_obj_FatFSFile_t *o = MP_OBJ_TO_PTR(self);
+  mp_buffer_info_t buf;
+  mp_get_buffer_raise(data, &buf, MP_BUFFER_WRITE);
+  UINT read;
+  FRESULT res = f_read(&(o->fp), buf.buf, buf.len, &read);
+  if (res < 0) {
```

Comment:
This condition will never be true, since res is unsigned. Should be `if (res)` like in the above cases.

Reply:
Fixed in cc5bb47d3f6d7178247271da6ffa5d0e97a5cfb8